kubeflow/core/nfs.libsonnet

```diff
@@ -0,0 +1,271 @@
+		},
+
+	    provisioner: {
+		  "apiVersion": "extensions/v1beta1", 
+		  "kind": "Deployment", 
+		  "metadata": {
+		    "name": provisionerName,	   
+		    "namespace": namespace,
+		  }, 
+		  "spec": {
+		    "replicas": 1, 
+		    "strategy": {
+		      "type": "Recreate"
+		    }, 
+		    "template": {
+		      "metadata": {
+		        "labels": {
+		          "app": provisionerName
+		        }
+		      }, 
+		      "spec": {
+		        "containers": [
+		          {
+		            "args": [
+		              "-provisioner=" + storageClassProvisioner,
+		            ], 
+		            "env": [
+		              {
+		                "name": "POD_IP", 
+		                "valueFrom": {
+		                  "fieldRef": {
+		                    "fieldPath": "status.podIP"
+		                  }
+		                }
+		              }, 
+		              {
+		                "name": "SERVICE_NAME", 
+		                "value": serviceName,
+		              }, 
+		              {
+		                "name": "POD_NAMESPACE", 
+		                "valueFrom": {
+		                  "fieldRef": {
+		                    "fieldPath": "metadata.namespace"
+		                  }
+		                }
+		              }
+		            ], 
+		            "image": "quay.io/kubernetes_incubator/nfs-provisioner:v1.0.8", 
+		            "imagePullPolicy": "IfNotPresent", 
+		            "name": "nfs-provisioner", 
+		            "ports": [
+		              {
+		                "containerPort": 2049, 
+		                "name": "nfs"
+		              }, 
+		              {
+		                "containerPort": 20048, 
+		                "name": "mountd"
+		              }, 
+		              {
+		                "containerPort": 111, 
+		                "name": "rpcbind"
+		              }, 
+		              {
+		                "containerPort": 111, 
+		                "name": "rpcbind-udp", 
+		                "protocol": "UDP"
+		              }
+		            ], 
+		            "securityContext": {
+		              "capabilities": {
+		                "add": [
+		                  "DAC_READ_SEARCH"
+		                ]
+		              }
+		            }, 
+		            "volumeMounts": [{
+		                // Needs to be mounted under /export because /export is what is exported for NFS.
+		                //    https://github.com/kubernetes-incubator/external-storage/tree/master/nfs#quickstart
+		                "mountPath": "/export", 
+		                "name": diskName,
+		             }],
+		          }
+		        ], 
+		        "volumes": [{
+		            "name": diskName,
+		            "gcePersistentDisk": {
+		              "pdName": diskName,
+		            },},],
+		        "serviceAccountName": serviceAccountName, 
+		      },
+		    },
+		  },
+		}, // provisioner
+    },
+
+    serviceAccount: {
+      "apiVersion": "v1", 
+      "kind": "ServiceAccount", 
+      "metadata": {
+        "labels": {
+          "app": name + "nfs-provisioner"
```

Comment:
Should the nfs-provisioner be its own component? Not sure if it is here already. I see it as  - if it launches a new (optional) component in the user's cluster, it should be its own component.

Reply:
Ideally and long term yes. I think the ideal way to do this would be to have distinct  packages for each thing; e.g a package for JupyterHub, a package for Spark, a package for the CRD etc...

We could then have one or more packages with one or more prototypes defining useful combinations of this components. Its not immediately clear what the organizing principle for those higher level packages should be.

In the short term though, its more convenient for users if they have fewer packages to install. So that was the motivation for putting them all in the same package.

Using more packages should be reasonable when ksonnet adds better support for managing package dependencies ; e.g. so that core can depend on some other packages and installing core automatically pulls in the other packages.